### ISSUE TITLE
One address that does not comply stops entire mailing

### ISSUE THREAD

**Issue by healthstatus** [mautic/mautic#1329]
I have tried to send an email, there is a non-compliant email address, and I get this error ERROR: [MAIL ERROR] Address in mailbox given [stupidperson.@yahoo.com] does not comply with RFC 2822

Mautic should skip it and go to the next email, not try 3 times every 15 minutes forever.

**Comment by alanhartless** [comment]
What context is the mail sent in? It shouldn't stop the entire mailing as Mautic should be catching the malformed email via exception, log it, and keep going.

**Comment by healthstatus** [comment]
It was a list email.  I have two days of server logs showing this error repeating and no emails got sent.

**Comment by alanhartless** [comment]
List emails are sent via ajax'ed batch process. Are you by chance familiar with reviewing the response of an ajax request through the browser you use? If so, could you post the response for when the sending seems to get stuck.

If the emails are being sent through a campaign, that's different than a list email and are executed through cron.  If you run the command manually through the console/SSH, what's the output?

**Comment by healthstatus** [comment]
I am familiar with Chrome developer tools, I go to the Network and see requests going through.  Ajax support on my list with 27,000+ on the list just doesn't make any sense.  I really want to press a button to send and not have to leave a browser window open for hours.  Even better I would like to schedule and send based on the recipients time zone.

**Comment by escopecz** [comment]
@healthstatus is the original issue solved? Your suggestions are great, but I'd create a separate issue for them.

**Comment by healthstatus** [comment]
No, I think about 5k of 28k went out, but the email is showing all 28k as unsent.

**Comment by PatchRanger** [comment]
Let me chip in with a bit different (at first glance) issue: my email queue suddenly got stuck with no error message anywhere at all - neither Mautic log, no PHP log. Quick investigation gave even more weird results:
- Corresponding cron job works as usual
- Running the same command via ssh just quickly returns without any error or output

I've finally caught the bug. It was in the condition before sanitizing queue. By "sanitizing" I mean, clearing out old messages that can't be sent and otherwise would (and are) blocking sending of "healthy" emails.
I am going to provide pull-request - let's see whether it fixes this issue. My own one was fixed by it.

**Comment by PatchRanger** [comment]
I've found another reason, which makes email queue stuck. Sometimes message couldn't be read: at first file_get_contents() silently fails (any code behind it won't get executed), then after I suppressed it using `@`, `unserialize()` silently fails - and can't be suppressed by `@`.
The reason is in the email body, which contains some characters, that are not correctly interpreted by `unserialize()`. So it silently fails - and other code not executed, which makes queue stuck.
As we have similar problems with serialization in #1029, I guess we should use either additional encoding before serialization (as described here: https://davidwalsh.name/php-serialize-unserialize-issues ) or just switch to json_encode/json_decode.
